packages/backend/src/scheduler/SchedulerClient.ts

```diff
@@ -186,18 +185,42 @@ export class SchedulerClient {
     ): Promise<void> {
         const dates = getDailyDatesFromCron(scheduler.cron);
         try {
-            const promises = dates.flatMap((date: Date) =>
-                scheduler.targets.map((target) =>
-                    this.addJob(date, scheduler, target),
-                ),
+            const promises = dates.map((date: Date) =>
+                this.addScheduledDeliveryJob(date, scheduler),
             );
 
             Logger.info(
-                `Creating new ${promises.length} notification jobs: ${scheduler.name}`,
+                `Creating ${promises.length} scheduled delivery jobs for scheduler ${scheduler.schedulerUuid}`,
             );
             await Promise.all(promises);
         } catch (err: any) {
-            Logger.error(`Unable to schedule job ${scheduler.name}`, err);
+            Logger.error(
+                `Unable to schedule job for scheduler ${scheduler.schedulerUuid}`,
+                err,
+            );
+            throw err;
+        }
+    }
+
+    async generateJobsForSchedulerTargets(
+        scheduler: SchedulerAndTargets,
+        page: NotificationPayloadBase['page'],
+    ): Promise<void> {
+        try {
+            const promises = scheduler.targets.map((target) =>
+                this.addNotificationJob(new Date(), scheduler, target, page),
+            );
+
+            Logger.info(
+                `Creating ${promises.length} notification jobs for scheduler ${scheduler.schedulerUuid}`,
+            );
+            await Promise.all(promises);
```

Comment:
So this fails the parent job if it can't schedule all of the jobs for the target?

Reply:
Is the idea that "not scheduling a job successfully" is actually quite unexpected. vs. a notification job failing (which might be expected depending on the delivery channel)